technic/machines/compat/digtron.lua

```diff
@@ -23,8 +79,28 @@ local function power_connector_compat()
 	})
 end
 
+local function battery_holder_compat()
+	-- Override battery holder
+	minetest.override_item("digtron:battery_holder",{
+		allow_metadata_inventory_put = function(pos, listname, index, stack, player)
+			-- Allow all items with energy storage from technic mod
+			-- And specifically if they hold any charge
+			-- Disregard empty batteries, the player should know better
+			if listname == "batteries" and technic.power_tools[stack:get_name()] then
+				return technic.get_RE_charge(stack) > 0 and stack:get_count() or 0
+			end
+			return 0
+		end,
+	})
+	-- Override digtron.tap_batteries
+	digtron.tap_batteries = tap_batteries
+end
```

Comment:
Something is not working here, battery holder does accept and reject correct items when placed by player but for some reason it does not work with pipeworks.
Battery holder is not accepting charged items from tubes, there might be some function caching after all.
Or pipeworks checks this different way.

This should be fixed somehow but first of course have to find why it does not work with pipeworks.

Reply:
Maybe it's because tubes use their own code... :wink:
https://github.com/minetest-mods/digtron/blob/master/nodes/node_battery_holder.lua#L101-L121

Reply:
Yes, have to add (clone?) function for `can_insert` and that should do it. 👍